cmd/agent/common/helpers.go

```diff
@@ -20,16 +22,24 @@ func SetupConfig(confFilePath string) error {
 
 // SetupConfigWithWarnings fires up the configuration system and returns warnings if any.
 func SetupConfigWithWarnings(confFilePath, configName string) (*config.Warnings, error) {
```

Comment:
Slightly off-topic, but isn't it more appropriate to call `SetupConfigWithWarnings` `SetupConfigWithSecrets` instead?